packages/core/primitives/event-dispatch/src/register_events.ts

```diff
@@ -6,34 +6,63 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {EarlyEventContract, EarlyJsactionDataContainer} from './earlyeventcontract';
 import {EventContractContainer} from './event_contract_container';
 import {EventContract} from './eventcontract';
 
+export type Tracker<T> = {[key: string]: {[appId: string]: T}};
+
 /**
  * Provides a factory function for bootstrapping an event contract on a
- * window object.
- * @param field The property on the window that the event contract will be placed on.
+ * specified object (by default, exposed on the `window`).
+ * @param field The property on the object that the event contract will be placed on.
  * @param container The container that listens to events
  * @param appId A given identifier for an application. If there are multiple apps on the page
  *              then this is how contracts can be initialized for each one.
  * @param events An array of event names that should be listened to.
- * @param anyWindow The global window object that should receive the event contract.
- * @returns The `event` contract. This is both assigned to `anyWindow` and returned for testing.
+ * @param earlyJsactionTracker The object that should receive the event contract.
  */
 export function bootstrapEventContract(
   field: string,
   container: Element,
   appId: string,
   events: string[],
-  anyWindow: any = window,
+  earlyJsactionTracker: any = window,
```

Comment:
Can we use `Tracker` type here as well?
```suggestion
  earlyJsactionTracker: Tracker<EventContract> = window,
```

Reply:
done